### PR TITLE
PMM-9077: Migrate DB before using it

### PR DESCRIPTION
**Pull request by oter**
[PMM-9077](https://jira.percona.com/browse/PMM-9077)

- [x] Build: https://github.com/Percona-Lab/pmm-submodules/pull/2108

- [ ] Links to other linked pull requests (optional).
---
- [x] Tests passed.
- [x] Feature build pass.
- [x] (Re)requested review.
- [x] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -735,11 +748,9 @@ func main() {
...
 			for {
 				deps.l.Warnf("Retrying in %s.", delay)
-				sleepCtx, sleepCancel := context.WithTimeout(ctx, delay)
-				<-sleepCtx.Done()
-				sleepCancel()
-
-				if ctx.Err() != nil {
+				select {
+				case <-time.After(delay):
+				case <-ctx.Done():
```
Comment (BupycHuk) [inline main.go:753]: why did you change this behaviour? it ignores ctx.Done()

File: main.go [hunk 2]
```diff
@@ -522,6 +507,35 @@ func getQANClient(ctx context.Context, sqlDB *sql.DB, dbName, qanAPIAddr string)
 	return qan.NewClient(conn, db)
 }
 
+func migrateDB(sqlDB *sql.DB, dbName, dbAddress, dbUsername, dbPassword string) {
+	l := logrus.WithField("component", "migration")
+
+	const timeout = 10 * time.Minute
+	timeoutCh := time.After(timeout)
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply main.go:514]: +1

**Review by BupycHuk — APPROVED** [review]
Where do we use DB before setup?
Could you, please create tech task to address @artemgavrilov's comment?

**Comment by artemgavrilov** [comment]
@BupycHuk  We load alerting templates from DB right before creating templates service. It happened before DB migrations.

File: main.go (hunk 1 above)
Comment (oter) [reply main.go:753]: ok, returned back

File: main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (oter) [reply main.go:514]: timeout is smaller and using ctx

**Review by BupycHuk — APPROVED** [review]
